streams/src/main/java/org/apache/kafka/streams/state/SessionBytesStoreSupplier.java

```diff
@@ -35,4 +35,11 @@
      * @return segmentInterval in milliseconds
      */
     long segmentIntervalMs();
+
+    /**
+     * The time period for which the {@link SessionStore} will retain historic data.
+     *
+     * @return retentionPeriod
+     */
+    long retentionPeriod();
```

Comment:
Seems to be a public API change?

Reply:
Good point, I think we should.

Could we piggy back it into KIP-319 (for connivence reasons only, I know it is not a very valid excuse)? @vvcephei @mjsax

Reply:
@guozhangwang I'm a little hesitant to do so in this case... This change, as well as the rationale and desired outcome, is really different than the one in 319, although they modify adjacent blocks of code.

As you say, it would be convenient as 319 only needs one more vote. But then again, if we change it, the folks who already voted would need to re-evaluate their decision anyway.

¯\_(ツ)_/¯ Call me "weakly opposed".

Reply:
Opposition well received. I will create a KIP for this then.